app/assets/javascripts/admin/index_utils/filters/attr_filter.js.coffee

```diff
@@ -1,12 +1,16 @@
 # Used like a regular angular filter where an object is passed
 # Adds the additional special case that a value of 0 for the filter
 # acts as a bypass for that particular attribute
+
+# NOTE the name doesn't reflect what the filter does, it only fiters on the variant.producer_id
 angular.module("admin.indexUtils").filter "attrFilter", ($filter) ->
   return (objects, filters) ->
-    Object.keys(filters).reduce (filtered, attr) ->
-      filter = filters[attr]
-      return filtered if !filter? || filter == 0
-      return $filter('filter')(filtered, (object) ->
-        object[attr] == filter
-      )
-    , objects
+    filter = filters["producer_id"]
+
+    return objects if !filter? || filter == 0
+
+    return $filter('filter')(objects, (product) ->
+      for variant in product.variants
+        return true if variant["producer_id"] == filter
+      false
+    , true)
```

Comment:
Took me a while to investigate to figure out what this is for. It appears to be for the Producer dropdown on the Inventory page, which filters the product list based on the selected supplier.

Now that product variants may come from different suppliers, the filter checks if any are from that supplier, and presumably _all_ variants for the product are shown (including ones from other suppliers). This won't be clear because the supplier is only shown on the product row.

_edit:_ oh, maybe you changed it to show on the variant row already.
But I think it might still show variants from other suppliers.

I think we want to phase out the Inventory screen, so perhaps this is an acceptable caveat, but one worth mentioning all the same.

Am  I on the right track?